### PR TITLE
asking the user to check their wallet when enabling it

### LINKED ISSUE #4707: Check your wallet!
When opening the checkout modal it is possible that the application might not have been "connected" by allowing it in the web3 wallet. If that's the case, we should show a message to the user asking them to check their wallet.

### PR DESCRIPTION
**Pull request by julien51**
# Description

This fixes #4707 by asking the user to check their wallet if needed.
This is far from elegant and will need tests and more.

# Issues

Fixes #4707

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/src/messageTypes.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export enum PostMessages {
   UPDATE_ACCOUNT = 'update/account',
   UPDATE_ACCOUNT_BALANCE = 'update/accountBalance',
   UPDATE_NETWORK = 'update/network',
-  UPDATE_WALLET = 'update/walletmodal',
+  UPDATE_WALLET = 'update/walletModal',
```
Comment (julien51) [inline paywall/src/messageTypes.ts]: just me being anal about camelCase and consistency...

File: paywall/src/components/content/CheckoutContent.tsx [hunk 2]
```diff
@@ -61,17 +62,24 @@ export default function CheckoutContent() {
...
     network,
     locks,
+    checkWallet,
   }: blockchainData = (useBlockchainData as useBlockchainDataFunc)(
     window,
     paywallConfig
   )
 
+  const [showWalletCheckOverlay, setShowWalletCheckOverlay] = useState(false)
+  const [walletOverlayDismissed, setWalletOverlayDismissed] = useState(false)
```
Comment (julien51) [inline paywall/src/components/content/CheckoutContent.tsx]: It's kind of gross but we want to only show the modal when `checkWallet` is set to true and the user has not explicitly dismissed it. After that, they're on their own.

File: paywall/src/unlock.js/CheckoutUIHandler.ts [hunk 3]
```diff
@@ -64,11 +64,6 @@ export default class CheckoutUIHandler {
       this.iframes.checkout.postMessage(PostMessages.UPDATE_NETWORK, network)
     )
 
-    // listen for wallet action
```
Comment (julien51) [inline paywall/src/unlock.js/CheckoutUIHandler.ts]: This was not used anywhere so cleaning it up!

File: paywall/src/unlock.js/PostMessageEmitters/CheckoutIframeMessageEmitter.ts [hunk 4]
```diff
@@ -42,29 +43,49 @@ export default class CheckoutIframeMessageEmitter extends FancyEmitter {
...
     const { postMessage, addHandler } = mainWindowPostOffice(
       window,
       this.iframe,
       url.origin,
       'main window',
       'Checkout UI iframe'
     )
-    this.postMessage = postMessage
+    // We want to only post message when we're ready!
+    this.postMessage = (type, payload) => {
```
Comment (julien51) [inline paywall/src/unlock.js/PostMessageEmitters/CheckoutIframeMessageEmitter.ts]: For now we are sending messages even if the iframe is not loaded. this logic adds sime kind of buffering to make sure that the iframe is ready before sending messages.

**Comment by cnasc** [comment]
This looks like it will fix the issue. I do think that the rendering logic for `CheckoutContent` will need to be decomposed into different parts for maintainability in future PRs, since it's rather convoluted.

**Comment by julien51** [comment]
Yeah agreed, this is getting gross.

File: paywall/src/__tests__/hooks/useBlockchainData.test.js [hunk 5]
```diff
@@ -25,22 +25,26 @@ describe('useBlockchainData hook', () => {
     )
   }
 
-  function getAddressListener() {
```
Comment (julien51) [inline paywall/src/__tests__/hooks/useBlockchainData.test.js]: The order of `fakeWindow.addEventListener.mock.calls` matters.

**Pull request merged**